modules/admin-ui/src/main/java/org/opencastproject/adminui/usersettings/UserSettingsService.java

```diff
@@ -217,20 +218,38 @@ private Function<EntityManager, UserSettings> getUserSettingsQuery(int limit, in
   public UserSetting addUserSetting(String key, String value) throws UserSettingsServiceException {
     String orgId = securityService.getOrganization().getId();
     String username = securityService.getUser().getUsername();
-    try {
-      return db.execTx(em -> {
-        UserSettingDto userSettingDto = new UserSettingDto();
-        userSettingDto.setKey(key);
-        userSettingDto.setValue(value);
-        userSettingDto.setUsername(username);
-        userSettingDto.setOrganization(orgId);
-        em.persist(userSettingDto);
-        return userSettingDto.toUserSetting();
-      });
-    } catch (Exception e) {
-      logger.error("Could not update user setting username '{}' org: '{}' key: '{}' value: '{}'", username, orgId, key,
-          value, e);
-      throw new UserSettingsServiceException(e);
+    Optional<UserSettingDto> userSettingOpt = db.exec(getUserSettingsByKeyQuery(key)).stream()
+        .filter(setting -> setting.getKey().equalsIgnoreCase(key))
+        .findFirst();
+    if (userSettingOpt.isPresent()) {
+      try {
+        return db.execTx(em -> {
+          UserSettingDto userSettingDto = userSettingOpt.get();
+          userSettingDto.setValue(value);
+          em.merge(userSettingDto);
+          return userSettingDto.toUserSetting();
+        });
+      } catch (Exception e) {
+        logger.error("Could not update user setting username '{}' org: '{}' key: '{}' value: '{}'", username, orgId,
+            key, value, e);
+        throw new UserSettingsServiceException(e);
+      }
+    } else {
+      try {
+        return db.execTx(em -> {
+          UserSettingDto userSettingDto = new UserSettingDto();
+          userSettingDto.setKey(key);
+          userSettingDto.setValue(value);
+          userSettingDto.setUsername(username);
+          userSettingDto.setOrganization(orgId);
+          em.persist(userSettingDto);
+          return userSettingDto.toUserSetting();
+        });
+      } catch (Exception e) {
+        logger.error("Could not add user setting username '{}' org: '{}' key: '{}' value: '{}'", username, orgId,
+            key, value, e);
+        throw new UserSettingsServiceException(e);
```

Comment:
Never log an error if you continue to throw the caught exception. If you want to include additional information, do that in the exception string of the new exception.

Reasoning:
- The thrown exception may be caught and handled later in which case the error is no longer an error, but expected and handled behavior.
- The exception may trickle up and be caught and logged there in which case you log the same error twice so that it looks like there are multiple erros. Especially on a busy system where other logs may show up in between.